### PR TITLE
Nana Library

### PR DESCRIPTION
**Pull request by kassane**
Ref: #6193

### CODE REVIEW

**Comment by kassane** [comment]
@Alexpux , what exactly is wrong? bug?

File: mingw-w64-nana/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
+        -DNANA_CMAKE_ENABLE_JPEG=YES \
+        -DNANA_CMAKE_ENABLE_PNG=YES \
+        -DNANA_CMAKE_INSTALL=YES \
+        -DNANA_CMAKE_NANA_FILESYSTEM_FORCE=NO \
+        -DNANA_CMAKE_STD_FILESYSTEM_FORCE=YES \
+        -B "${_builddir}-${MINGW_CHOST}"
+    
+    cmake --build "${_builddir}-${MINGW_CHOST}"
```
Comment (Alexpux) [inline mingw-w64-nana/PKGBUILD]: Use make

File: mingw-w64-nana/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Maintainer: <matheus-catarino@hotmail.com>
+
+_realname=nana
+_builddir=binary
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=1.7.3
+pkgrel=1
+pkgdesc="An opensource cross-platform GUI library written in modern C++11 for static linking(mingw-w64)"
```
Comment (Alexpux) [inline mingw-w64-nana/PKGBUILD]: Add space before ````(mingw-w64)````

File: mingw-w64-nana/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        -DNANA_CMAKE_NANA_FILESYSTEM_FORCE=NO \
+        -DNANA_CMAKE_STD_FILESYSTEM_FORCE=YES \
+        -B "${_builddir}-${MINGW_CHOST}"
+    
+    cmake --build "${_builddir}-${MINGW_CHOST}"
+}
+
+package() {
+    cd ${srcdir}/${_realname}-${pkgver}
+    cmake --install "${_builddir}-${MINGW_CHOST}" --prefix "${pkgdir}/${MINGW_PREFIX}"
```
Comment (Alexpux) [inline mingw-w64-nana/PKGBUILD]: Use make

File: mingw-w64-nana/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+source=("${_realname}-v${pkgver}.tar.gz"::"https://github.com/cnjinhao/${_realname}/archive/v${pkgver}.tar.gz")
+sha256sums=('fb2480477dc48fbbbbe4a12a07b67d775c32516c5c2e8bb5b6a95361d3b07585')
+
+build() {
+    cd ${srcdir}/${_realname}-${pkgver}
+     MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+        ${MINGW_PREFIX}/bin/cmake \
+        -G"MSYS Makefiles" \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${pkgdir}${MINGW_PREFIX} \
```
Comment (Alexpux) [inline mingw-w64-nana/PKGBUILD]: Remove pkgdir here

File: mingw-w64-nana/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        -DNANA_CMAKE_ENABLE_JPEG=YES \
+        -DNANA_CMAKE_ENABLE_PNG=YES \
+        -DNANA_CMAKE_INSTALL=YES \
+        -DNANA_CMAKE_NANA_FILESYSTEM_FORCE=NO \
+        -DNANA_CMAKE_STD_FILESYSTEM_FORCE=YES \
+        -B "${_builddir}-${MINGW_CHOST}"
+    
+    cd ${srcdir}/${_realname}-${pkgver}/${_builddir}-${MINGW_CHOST}
```
Comment (Alexpux) [inline mingw-w64-nana/PKGBUILD]: Please look at other our PKGBUILDs and see how it done there

File: mingw-w64-nana/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    make
+}
+
+package() {
+    cd ${srcdir}/${_realname}-${pkgver}/${_builddir}-${MINGW_CHOST}
+    make install DESTDIR="${pkgdir}" prefix="${MINGW_PREFIX}"
```
Comment (Alexpux) [inline mingw-w64-nana/PKGBUILD]: See our PKGBUILDs

Comment (kassane) [reply mingw-w64-nana/PKGBUILD]: Before it was also with the zeal package, using cmake directly and builded normally. Now I'm adapting for make. A suggestion could perhaps be clearer.

Comment (Alexpux) [reply mingw-w64-nana/PKGBUILD]: @kassane take a look into your zeal package that i fix myself and do the same

**Comment by Alexpux** [comment]
@kassane use make as I suggested. I will not accept your packages while you not folow our syntax

**Comment by kassane** [comment]
> 
> 
> @kassane use make as I suggested. I will not accept your packages while you not follow our syntax

Ok

File: mingw-w64-nana/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        -DNANA_CMAKE_INSTALL=YES \
+        -DNANA_CMAKE_NANA_FILESYSTEM_FORCE=NO \
+        -DNANA_CMAKE_STD_FILESYSTEM_FORCE=YES
+    
+    make
+}
+
+package() {
+    cd ${srcdir}/${_realname}-${pkgver}
+    make install DESTDIR="${pkgdir}" prefix="${MINGW_PREFIX}"
```
Comment (Alexpux) [inline mingw-w64-nana/PKGBUILD]: You dont need pass prefix there

**Pull request merged**